src/user/user.service.ts

```diff
@@ -45,6 +45,38 @@ export class UserService {
     return user;
   }
 
+  public async changePassword(
+    userId: number,
+    currentPassword: string,
+    newPassword: string,
+  ): Promise<void> {
+    const user = await this.repository.findOne({ id: userId });
+
+    if (!user) {
+      throw new Error('Benutzer nicht gefunden.');
+    }
+
+    // Überprüfe das aktuelle Passwort
```

Comment:
Die Kommentare in dieser Methode sind eigentlich überflüssig, der Code, bzw. auch die Variablennamen sind ja selbsterklärend. Könnte man also noch entfernen, sonst passt das aber.